pkg/refactor/core/gateway.go

```diff
@@ -2,20 +2,31 @@ package core
 
 import (
 	"net/http"
-	"sync"
 )
 
 const MagicalGatewayNamespace = "gateway_namespace"
 
-type GatewayManager interface {
+type EndpointManager interface {
 	http.Handler
 
-	ListEndpoints() []*Endpoint
-	SetEndpoints(endpoints []*Endpoint)
-
-	Start(done <-chan struct{}, wg *sync.WaitGroup)
+	GetAll() []*EndpointStatus
+	SetEndpoints(endpoints []*Endpoint) []*EndpointStatus
 }
 
 type Endpoint struct {
-	Method string `json:"method"`
+	Method    string    `json:"method"`
+	FilePath  string    `json:"file_path"`
+	Workflow  string    `json:"workflow"`
+	Namespace string    `json:"namespace"`
```

Comment:
I think we agreed that `Workflow` and `Namespace` should be part of plugin config and not directly under the endpoint config.